Makefile

```diff
@@ -1,23 +1,38 @@
-JS := $(shell find . -name '*.mjs' | sed 's/\.mjs/\.js/')
+# If there's an activated virtualenv, use that. Otherwise, make one in the cwd.
+# We avoid $(CURDIR) because it spits out /cygdrive/c/... on Windows Cygwin
+# installs and leads to things that don't work.
+VIRTUAL_ENV ?= ./venv
+PYTHON3 ?= python3
+PATH := ./node_modules/.bin:$(VIRTUAL_ENV)/bin:$(VIRTUAL_ENV)/Scripts:$(PATH)
+
+JS := $(shell find . -name '*.mjs' | grep -v '^./node_modules/.*' | sed 's/\.mjs/\.js/')
 
 # It's faster to invoke Babel once and compile everything than to invoke it
 # separately on even 2 individual files that changed.
-%.js: %.mjs; @node_modules/.bin/babel *.mjs **/*.mjs --out-dir .
+%.js: %.mjs .npm_installed; @node_modules/.bin/babel *.mjs **/*.mjs --out-dir .
 
 all: $(JS)
 
-lint:
+lint: js_lint py_lint
+
+js_lint: $(JS)
 	@node_modules/.bin/eslint --ext mjs .
 	@node_modules/.bin/eslint test/browser
 
-# Does not include mocha browser tests. Use 'npm test' (Linux-based shells only) to run all mocha tests.
-test: $(JS)
-	@node_modules/.bin/mocha
-	pytest cli/fathom_web/test
+py_lint: $(VIRTUAL_ENV)/pyvenv.cfg
+	flake8 cli
+
+test: js_test py_test
 
-coverage: $(JS)
+js_test: $(JS)
 	@node_modules/.bin/istanbul cover node_modules/.bin/_mocha -- --recursive
```

Comment:
Windows requires: `@node_modules/.bin/istanbul cover node_modules/mocha/bin/_mocha -- --recursive` as noted [here](https://github.com/gotwarlost/istanbul#usage-on-windows). This was a problem I ran into when I first started and had as a local change and completely forgot about.

Reply:
Huh. Then I'm surprised .bin/eslint and .bin/babel work. Do they? They're all symlinks on my box.

Reply:
Ah, reading the link made the reasons clear.